google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardLibraryPanel.java

```diff
@@ -100,21 +127,43 @@ public boolean isEnabled() {
     return enabled;
   }
 
-  @SuppressWarnings("checkstyle:abbreviationaswordinname")
-  private void createUIComponents() {
-    libraryPanel = new JPanel(new GridLayout(AppEngineStandardMavenLibrary.values().length, 1));
-    if (enabled) {
-      libraryPanel.setBorder(
-          BorderFactory.createTitledBorder(BorderFactory.createEtchedBorder(), "Libraries"));
-
-      for (AppEngineStandardMavenLibrary library : AppEngineStandardMavenLibrary.values()) {
-        final JCheckBox libraryCheckbox = new JCheckBox(library.getDisplayName());
-        libraryPanel.add(libraryCheckbox);
-
-        // The Servlet API is provided by the AE standard runtime. So we are enabling it by default.
-        if (library == AppEngineStandardMavenLibrary.SERVLET_API) {
-          libraryCheckbox.setSelected(true);
-        }
+  @VisibleForTesting
+  JCheckBox getLibraryCheckbox(String name) {
+    for (JCheckBox libraryCheckbox : libraries) {
+      if (name.equals(libraryCheckbox.getText())) {
+        return libraryCheckbox;
+      }
+    }
+
+    return null;
+  }
+
+  @VisibleForTesting
+  List<JCheckBox> getLibraries() {
+    return libraries;
+  }
+
+  private class AppEngineApiListener implements ItemListener {
```

Comment:
AppEngineApiCheckboxListener? is this actually used more than once? Maybe you can inline it?